lib/presentation/widget/shared_space_details/add_shared_space_member/add_shared_space_member_widget.dart

```diff
@@ -0,0 +1,267 @@
+            ],
+          )
+        ]));
+  }
+
+  Widget _membersListWidget(List<SharedSpaceMember> members) {
+    return Container(
+      color: AppColor.topBarBackgroundColor,
+      child: Theme(
+          data: ThemeData().copyWith(dividerColor: Colors.transparent),
+          child: ExpansionTile(
+            initiallyExpanded: true,
+            title: Padding(
+              padding: EdgeInsets.only(left: 8),
+              child: Text(AppLocalizations.of(context).existing_members(members.length),
+                  style: TextStyle(color: AppColor.addSharedSpaceMemberTitleColor, fontSize: 16)),
+            ),
+            children: [
+              Container(
+                  height: 400.5,
+                  color: Colors.white,
+                  child: ListView.builder(
+                      shrinkWrap: true,
+                      itemCount: members.length,
+                      itemBuilder: (context, index) {
+                        return SharedSpaceMemberListTileBuilder(
+                                members[index].account.name,
+                                members[index].account.mail,
+                                members[index].role.name.getRoleName(context),
+                                tileColor: Colors.white)
+                            .build();
+                      }))
+            ],
+          )),
+    );
+  }
+
+  void selectRoleBottomSheet(BuildContext context) {
+    showModalBottomSheet(
+        shape: RoundedRectangleBorder(
+            borderRadius:
+                BorderRadius.only(topLeft: Radius.circular(20.0), topRight: Radius.circular(20.0))),
+        context: context,
+        builder: (BuildContext context) {
+          return StatefulBuilder(builder: (BuildContext context, StateSetter setState) {
+            return Container(
+                child: Padding(
+                    padding: EdgeInsets.all(20),
+                    child: Column(
+                      mainAxisSize: MainAxisSize.min,
+                      children: [
+                        _getSelectingRoleTile(SharedSpaceRoleName.READER),
+                        _getSelectingRoleTile(SharedSpaceRoleName.ADMIN),
+                        _getSelectingRoleTile(SharedSpaceRoleName.CONTRIBUTOR),
+                        _getSelectingRoleTile(SharedSpaceRoleName.WRITER),
+                      ],
+                    )));
+          });
+        });
+  }
+
+  ListTile _getSelectingRoleTile(SharedSpaceRoleName role) {
+    return ListTile(
+      title: Text(role.getRoleName(context),
+          style: TextStyle(
+            color: _model.selectedRole == role
+                ? AppColor.primaryColor
+                : AppColor.addSharedSpaceMemberRoleTileColor,
+            fontSize: 16,
+          )),
+      leading: _model.selectedRole == role
+          ? Icon(Icons.check, color: AppColor.primaryColor)
+          : SizedBox(),
+      onTap: () => setState(() => _model.selectRole(role)),
```

Comment:
so weird to me too.

Reply:
I DON'T LIKE `setState()`. Please minimize to use that. @manhduc-nguyen @imGok @huynguyennovem @dab246

Reply:
We can not avoid it? @imGok